pydantic/_internal/_generate_schema.py

```diff
@@ -1105,6 +1151,10 @@ def set_discriminator(schema: CoreSchema) -> CoreSchema:
         schema = self._apply_field_serializers(
             schema, filter_field_decorator_info_by_field(decorators.field_serializers.values(), name)
         )
+        field_title_generator = field_info.field_title_generator or self._config_wrapper.field_title_generator
+        if field_title_generator is not None:
+            self._apply_field_title_generator_to_field_info(field_title_generator, field_info, name)
+
```

Comment:
Looks like we repeat this exact same logic 3x at least, maybe use a function to abstract away some of this?